shoulder/generator/c_header_generator.py

```diff
@@ -21,463 +21,767 @@
 # SOFTWARE.
 
 import os
+import textwrap
 
 from shoulder.generator.abstract_generator import AbstractGenerator
-from shoulder.model.register import Register
 from shoulder.logger import logger
 from shoulder.config import config
-from shoulder.exception import *
-from shoulder.gadget import *
+from shoulder.exception import ShoulderGeneratorException
+from shoulder.filter import filters
+from shoulder.transform import transforms
+import shoulder.gadget
+
 
 class CHeaderGenerator(AbstractGenerator):
     def generate(self, objects, outpath):
         try:
+            regs = objects
+
+            regs = transforms["remove_reserved_0"].transform(regs)
+            regs = transforms["remove_reserved_1"].transform(regs)
+            regs = transforms["remove_reserved_sign_extended"].transform(regs)
+            regs = transforms["remove_implementation_defined"].transform(regs)
+            regs = transforms["special_to_underscore"].transform(regs)
+            regs = transforms["remove_coprocessor_am"].transform(regs)
+            regs = transforms["remove_memory_mapped_am"].transform(regs)
+            regs = transforms["remove_system_vector_am"].transform(regs)
+            regs = transforms["remove_system_banked_am"].transform(regs)
+            regs = transforms["remove_system_immediate_am"].transform(regs)
```

Comment:
I imagine that these mechs are being removed since we currently aren't supporting them in our generator until we can figure out how to write the accessors?

Reply:
Yes, that is right. I'm glad you called out this point though because I think we should come up with a plan to parallelize ourselves to add support for each different type. What do you think about this plan:

1. I can do a quick PR to remove all of the accessor macros that aren't being used any more. 
2. Lets open individual issues on our issue tracker for each new access mechanism
3. For each access mechanism on the issue tracker: figure out how to make an accessor macro, fill in the appropriate get/set functions (just replace the "TODO" message), and then remove the corresponding filter.

Reply:
Yeah, that sounds like a good plan to me. Right now this is good because we have the placeholders to remind us, but it's still not going to mess with what we already have developed.